packages/synapse-react-client/src/components/EntitySubjectsSelector/EntitySubjectsSelector.tsx

```diff
@@ -0,0 +1,74 @@
+import { Box, Typography } from '@mui/material'
+import {
+  Reference,
+  ReferenceList,
+  RestrictableObjectDescriptor,
+  RestrictableObjectType,
+} from '@sage-bionetworks/synapse-types'
+import { isEqual, uniqWith } from 'lodash-es'
+import React, { useMemo } from 'react'
+import EntityHeaderTable, { EntityHeaderTableProps } from '../EntityHeaderTable'
+
+export const REMOVE_BUTTON_TEXT = 'Mark for Removal from AR'
+export const NO_ENTITIES_SELECTED = 'No entities selected.'
+
+export type EntitySubjectsSelectorProps = {
+  // will be filtered to only entity subjects
+  subjects: RestrictableObjectDescriptor[]
+  onUpdate?: (subjects: RestrictableObjectDescriptor[]) => void
+  onUpdateEntityIDsTextbox: EntityHeaderTableProps['onUpdateEntityIDsTextbox']
+}
+
+const EntitySubjectsSelector: React.FunctionComponent<
+  EntitySubjectsSelectorProps
+> = (props: EntitySubjectsSelectorProps) => {
+  const { subjects, onUpdate, onUpdateEntityIDsTextbox } = props
+
+  const references = useMemo(() => {
+    return subjects
+      .filter(subject => {
+        return subject.type === RestrictableObjectType.ENTITY
+      })
+      .map(subject => {
+        const ref: Reference = {
+          targetId: subject.id,
+        }
+        return ref
+      })
+  }, [subjects])
+
+  const handleChange = (updatedReferences: ReferenceList) => {
+    if (onUpdate) {
+      const dedupedReferences = uniqWith(updatedReferences, isEqual)
```

Comment:
`EntityHeaderTable` allows the user to add duplicate entities, but AR should not contain duplicate subjects, so drop duplicate entities as originally noted in SWC-3585 and [SWC test previously confirmed](https://github.com/Sage-Bionetworks/SynapseWebClient/blob/9085d9d2f9644ba7ceb9db9624b3a48e52a269a4/src/test/java/org/sagebionetworks/web/unitclient/widget/accessrequirements/createaccessrequirement/CreateAccessRequirementStep1Test.java#L305-L308)